.github/workflows/build-package.yml

```
@@ -0,0 +1,73 @@
+name: Build Package
+
+env:
+  GH_TOKEN: "${{ secrets.GH_TOKEN }}"
+  PACKAGING_SECRET_KEY: "${{ secrets.PACKAGING_SECRET_KEY }}"
+  PACKAGE_CLOUD_REPO_NAME: "sample"
+  PACKAGE_ENCRYPTION_KEY: "${{ secrets.PACKAGE_ENCRYPTION_KEY }}"
+  PACKAGING_PASSPHRASE: "${{ secrets.PACKAGING_PASSPHRASE }}"
+  DOCKERHUB_PASSWORD: "${{ secrets.DOCKERHUB_PASSWORD }}"
+  DOCKERHUB_USER_NAME: "${{ secrets.DOCKERHUB_USER_NAME }}"
+  CURRENT_BRANCH: ${GITHUB_REF##*/}
+on:
+  push:
+    branches: "**"
+
+  workflow_dispatch:
+
+jobs:
+  build_package:
+    name: Build package
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        TARGET_PLATFORM:
+          - centos,8
+          - centos,7
+          - debian,bullseye
+          - debian,buster
+          - debian,stretch
+          - oraclelinux,8
+          - oraclelinux,7
+          - ubuntu,focal
+          - ubuntu,bionic
+          - ubuntu,xenial
+          - pgxn
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install package dependencies
+        run: sudo apt-get update && sudo apt-get install libcurl4-openssl-dev libssl-dev python3-testresources
+
+      - name: Install wheel for el/8
+        if: matrix.TARGET_PLATFORM  == 'centos,8'
+        run: python -m pip install wheel
+
+      - name: Update and check dockerfiles
+        run: |
+          ./update_dockerfiles
```

I have 2 concerns here.
1. We do not need to run this script for every OS in the matrix. One is enough.
2. Since we do not fail fast, the test outputs are not clear if this step fails.

Can we somehow change the structure here a little bit?

---

1. Created a job to assure this step executes once
2. https://github.com/citusdata/packaging/blob/56870d2c26e47b9b76a73701a953761e1f9c39f4/update_dockerfiles#L4
Script is failing fast and Github Actions fails fast by default so

---

What about Line 34 (`fail-fast: false`) then? Does that not mean anything?

https://github.com/citusdata/packaging/pull/745/files#diff-c37cf2bffb2cde25cbcedc64387a46a1b752a13b0f310ecc68058b87889a3ac0R34

---

https://www.edwardthomson.com/blog/github_actions_6_fail_fast_matrix_workflows.html
It is about matrix which enables all the other jobs continue in the matrix even if one of them fails